seismicpro/src/seismic_batch.py

```diff
@@ -1253,3 +1257,44 @@ def equalize(self, index, src, dst, params, survey_id_col=None):
 
         getattr(self, dst)[pos] = equalized_field
         return self
+
+    @action
+    @inbatch_parallel(init='_init_component', target="threads")
+    def shift_pick_phase(self, index, src, src_traces, dst=None, shift=1.5*np.pi, threshold=0.05):
+        """ Shifts picking time stored in `src` component on the given phase along the traces stored in `src_raw`.
+
+        Parameters
+        ----------
+        src : str
+            The batch components to get the data from.
+        dst : str
+            The batch components to put the result in.
+        src_traces: str
+            The batch components where the traces are stored.
+        shift: float
+            The amount of phase to shift, default is 1.5 * np.pi which corresponds to transfering picking times
+            from 'max' to 'zero' type.
+        threshold: float
+            Threshold determining how many trace samples with low amplitudes, less then thd, can be skipped.
+            Introduced because of the unstable behaviour of the hilbert transform at the begining of the signal.
+
+         """
+        pos = self.get_pos(None, src, index)
+        pick = getattr(self, src)[pos]
+        trace = getattr(self, src_traces)[pos]
+        if isinstance(self.index, KNNIndex):
+            trace = trace[0]
+        trace = np.squeeze(trace)
+
+        analytic = hilbert(trace)
+        phase = np.unwrap(np.angle(analytic))
+        # finding x such that phase[x] = phase[pick] - shift
+        phase_mod = phase - phase[pick] + shift
```

Comment:
```suggestion
        phase_mod = phase - (phase[pick] - shift)
```
Just a bit more self-explanatory